server/server/Controllers/AdminController.cs

```diff
@@ -68,23 +46,19 @@ public IActionResult Post(AddedCake cake)
 
             try
             {
-                string path, 
-                    cakePhotoName = cake.Img.FileName, 
+                string path,
+                    cakePhotoName = Guid.NewGuid().ToString() + cake.Img.FileName,
                     folderPath = Directory.GetCurrentDirectory() + "wwwroot" + "images";
-                if (System.IO.File.Exists(Path.Combine(Directory.GetCurrentDirectory(), "wwwroot", "images", cakePhotoName)))
-                {
-                    cakePhotoName = DateTime.Now.ToString("MMddHHmmss") + cake.Img.FileName;
-                }
                 path = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot", "images", cakePhotoName);
                 using (Stream stream = new FileStream(path, FileMode.Create))
                 {
                     cake.Img.CopyTo(stream);
                 }
-                List<User> users = db.Users.GetAll().ToList();
+                List<User> users = db.Users.GetAll().Where(u => !u.IsAdmin).ToList();
```

Comment:
For this app it is ok, but better to run filtration on db. In case if there is a lot of users they all will be retrieved that can take time